### ISSUE TITLE
CMake stdout build instructions bug?

### ISSUE THREAD

**Issue by johnomotani** [boutproject/BOUT-dev#2472]
If I configure with, for example,
```
cmake . -B build -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_COMPILER=CC -DBOUT_DOWNLOAD_SUNDIALS=ON
```
(i.e., as suggested by the manual, without `cd`'ing to `build`), I get the instruction printed at the end of a successful configuration
```
*** Now run `cmake --build .` to compile BOUT++ ***

-- Configuring done
-- Generating done
-- Build files have been written to: ...
```
but `cmake --build .` will not work - I need `cmake --build build`.

I don't know how to make the suggestion always correct (I guess it would at least need to detect whether you'd done `cmake . -B build` or `mkdir build; cd build; cmake ..`, and the directory might be called something other than `build`...), but it's not great that the current instruction can be incorrect. One quick fix might be to just link back to the docs (https://bout-dev.readthedocs.io/en/latest/user_docs/installing.html#cmake) instead of giving a specific command?

**Comment by dschwoerer** [comment]
Already fixed in next:
https://github.com/boutproject/BOUT-dev/blob/next/CMakeLists.txt#L1020

**Comment by dschwoerer** [comment]
Maybe worth back-porting to master?

**Comment by ZedThree** [comment]
Looks like this is fixed in next, but not 4.4. In next we have

```cmake
*** Now run `cmake --build ${CMAKE_BINARY_DIR}` to compile BOUT++ ***
```

which gives an absolute path. Easy enough to backport.

Annoyingly, I don't think there's a way to make that into a nice relative path, as I'm not sure it's possible to get the path where `cmake` is being invoked from

**Comment by dschwoerer** [comment]
I think cmake doesn't change the working directory, so calling `$(pwd)` or similar would give you the invoking directory.
